pennylane/math/multi_dispatch.py

```diff
@@ -896,3 +896,32 @@ def detach(tensor, like=None):
         return np.to_numpy(tensor)
 
     return tensor
+
+
+@multi_dispatch()

```

Comment:
I don' think the multi dispatch is necessary as you have only one arg of interest (array) otherwise you can set `@multi_dispatch(argnum=[0], tensors_list=[0])`